### PR TITLE
SL-5978 Create escalation logger class

### PR DESCRIPTION
**Pull request by eddspencer**
### Description of Changes

Create a new class that extends `Logger` so that we can easily add escalation options to our logs. This PR includes the basic structure and examples with mocks in tests - if this is agreed will make some helpful implementations of the trigger and action interfaces as well in a separate PR.

### Documentation

N/A

### Risks & Impacts

Low risk this is adding new functionality

### Testing
Added unit tests.

## Final Checklist

Please tick once completed.

- [x] Build passes.
- [x] Versioning considered (the version number in this PR is inline with semantic 
versioning requirements).
- [x] Change log has been updated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ structured arguments to be included in `ch.qos.logback.classic.net.SyslogAppende
 <dependency>
   <groupId>com.echobox</groupId>
   <artifactId>ebx-structuredlogging-sdk</artifactId>
-  <version>1.0.0</version>
```
Comment (jack7ellis) [inline README.md:20]: I feel like this is worth of a minor version bump

File: src/main/java/com/echobox/logging/escalator/LoggingLevel.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
```
Comment (jack7ellis) [inline src/main/java/com/echobox/logging/escalator/LoggingLevel.java:1]: surprised that this enum doesn't already exist in slf4j

File: src/main/java/com/echobox/logging/escalator/EscalationLogger.java [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+  public boolean isInfoEnabled() {
+    return logger.isInfoEnabled();
+  }
+  
+  @Override
+  public boolean isInfoEnabled(Marker marker) {
+    checkForEscalation(LoggingLevel.INFO, marker);
+    
+    return logger.isInfoEnabled(marker);
+  }
```
Comment (jack7ellis) [inline src/main/java/com/echobox/logging/escalator/EscalationLogger.java]: Out of interest, why do we check for escalation in these methods?

**Review by jack7ellis — APPROVED** [review]
Looks like a good approach Edd. Looking forward to seeing it in action.

File: src/main/java/com/echobox/logging/escalator/EscalationLogger.java (hunk 3 above)
Comment (eddspencer) [reply src/main/java/com/echobox/logging/escalator/EscalationLogger.java]: oh sorry that is a copy/paste issue - got a bit carried away!

File: src/main/java/com/echobox/logging/escalator/LoggingLevel.java (hunk 2 above)
Comment (eddspencer) [reply src/main/java/com/echobox/logging/escalator/LoggingLevel.java:1]: me 2!

**Pull request merged**